### PR TITLE
[WIP] As a player in a multiplayer game I should be able to click on a url in a message to open that url in my browser.

### PR DESCRIPTION
**Pull request by jasper-lyons**
Takes up the work of: https://github.com/OpenRA/OpenRA/pull/6365

This PR rebases #6365 onto bleed. All of the logic and functionality is updated to work as in the original PR but this is just a starting point.

There are some decisions that need to be made:

* Whether to change track and implement selecting and copying text (IMO, this solution is preferable because it avoids owning possible attack / spam vectors).

* How restruictive to be with urls that can be clicked.

* What color to indicate clickable links (IMO we should jsut use web default styles, underline + #0000EE, we shouldn't attempt to implement a visited state, OpenRA isn't a browser).

* Which cursor types to use when mousing over clickable text.

There are some things I think would be worth considering at a technical level:

* If we have interactable text we should move this into it's own widget and give the parent module the responcibility of parseing the message, breaking it into separate labels and using something like a URLLabelWidget for the urls and a simple LabelWidget for the non-interactive text. What does anyone else think?

References:
https://github.com/OpenRA/OpenRA/issues/7517
Closes https://github.com/OpenRA/OpenRA/issues/6857.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/LabelWidget.cs [hunk 1]
```diff
@@ -107,6 +144,52 @@ public override void Draw()
...
+				// for our solution. We should think about how we're going to do this more as
+				// it presents an attack vector for maliciously sending users to bad urls or
+				// running arbitrary code on their machines e.g.
+				//
+				// Process.Start("https://google.com; nc -l 8000 | sh"r
+				switch (OpenRA.Platform.CurrentPlatform)
+				{
+					case OpenRA.PlatformType.Windows:
+						string url = ClickURL.Replace("&", "^&");
+						Process.Start(new ProcessStartInfo("cmd", "/c start " + url) { CreateNoWindow = true });
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/LabelWidget.cs]: That is not URL clicking, that is remote code execution!

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/LabelWidget.cs]: I see that this is a workaround for https://github.com/dotnet/corefx/issues/10361 but it doesn't really look safe.

Comment (jasper-lyons) [reply OpenRA.Mods.Common/Widgets/LabelWidget.cs:170]: Yup. It isn't and I'm not happy with it either. I'm moving to allow people to select and copy arbitrary text instead. Then we're not in charge of this, just enabling them to copy the url.

**Comment by pchote** [comment]
@jasper-lyons what would you like us to do wrt this PR?

File: OpenRA.Mods.Common/Widgets/LabelWidget.cs (hunk 1 above)
Comment (teinarss) [reply OpenRA.Mods.Common/Widgets/LabelWidget.cs:170]: Should work when we are done with the migration to core 3.1. See https://github.com/dotnet/corefx/pull/23705

**Comment by jasper-lyons** [comment]
I haven't had the time to continue with it for a while. I will push my progress. I should be able to focus on it again in a few weeks. Wish there was a "paused" concept in github prs.

**Comment by RobFog** [comment]
I would welcome this.

The pull request needs to be rebased.

**Comment by Mailaender** [comment]
I'd say this needs to wait for #17989 to work safely on Windows.

**Comment by jasper-lyons** [comment]
I moved to a different approach and would like to close this in favour of https://github.com/OpenRA/OpenRA/pull/18404

Does anyone have any reasons not to do this? I will close this very soon otherwise.

**Comment by teinarss** [comment]
@jasper-lyons are you interested to resurrect this? We have merged #17989 and another option is to use SDL_OpenURL.

**Pull request closed without merging**